ext/standard/http.c

```diff
@@ -27,24 +27,17 @@ static void php_url_encode_scalar(zval *scalar, smart_str *form_str,
 	const char *index_string, size_t index_string_len,
 	const char *num_prefix, size_t num_prefix_len,
 	const zend_string *key_prefix,
-	const zend_string *arg_sep)
```

Comment:
Can you explain why passing a char* + size_t is better?

Reply:
It's not that it's better than passing a `zend_string*`, it's that we have no choice here ;)
In the profiler I saw that `zend_ini_str` took 6% of time because it has to go through a bunch of lookups via hashtables and some extra logic, whereas we could just grab the separator from the PG globals. However, they're stored as `char*` in the globals instead of as `zend_string*`, so this forces me to use a `char*`+`size_t` combination. Effectively this also has the "advantage" that if you want to pass a literal you can just do so without creating a `zend_string` instance first.

Reply:
It should be easy enough to store the arg separator as `zend_string*` by using `OnUpdateStrUnempty` instead of `OnUpdateStringUnempty`. That would also avoid quite a few `strlen()` calls.